Enterprise-CMCS/macpro-mdct-mcr
# Playwright setup

### Description

Most of this is copied from MFP, with minor changes for repository naming and organization

- Add new env variables for playwright
  - these use the same cypress credentials for now, but sets us up to swap them later if need be
- Add cron job to remove old playwright reports from `gh-pages` branch
  - Add rm_old_folders.py script for cron job
- Add playwright tests and report upload to deploy workflow
- Add playwright dependencies in test package
- Add playwright config and base setup and tests from MFP
  - These are almost unchanged from MFP, with a few changes to update expected headings

### Related ticket(s)


CMDCT-4127

---
### How to test

- View the deploy workflow for this branch
  - Verify it runs playwright tests after cypress
  - Verify it won't delist the github actions cidrs until tests are done
  - Verify it uploads reports by viewing the github pages link provided in the summary

Pull branch and run playwright tests locally
`cd tests/ && yarn`
`playwright install browsers`
`yarn test:e2e-ui`

### Notes




---
### Pre-review checklist
